swift/Workflow/Sources/SideEffect.swift

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2020 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+public protocol SideEffect: Hashable {
+    associatedtype Action: WorkflowAction
+
+    func run(sink: Sink<Action>) -> Lifetime
+}
```

Comment:
Squinting, this looks a lot like `Worker`.

Reply:
It pretty much is - except we're moving the `ReactiveSwift` specific stuff out of `Worker`. So instead of it returning `SignalProducer<Output>`, we'll be passing in a `Sink` and use an internal concept for `Lifetime` instead.

Oh, we're also adding a `hash` value to it.